cockpit/src/main/java/com/polidea/cockpit/manager/CockpitParam.kt

```diff
@@ -1,3 +1,15 @@
 package com.polidea.cockpit.manager
 
-data class CockpitParam(val name: String, val typeClass: Class<*>, var value: Any)
\ No newline at end of file
+import com.polidea.cockpit.listener.CockpitParamChangeListener
+import kotlin.properties.Delegates
+
+data class CockpitParam<T>(val name: String, val typeClass: Class<*>, private var _value: T) {
+
+    var cockpitParamChangeListener: CockpitParamChangeListener<T>? = null
+
+    var value: T by Delegates.observable(_value) { _, old, new ->
+        _value = new
+        if (old != null && new != null && old != new)
```

Comment:
What will happen if `old` value is `null` and `new` value will be `not-null`? Probably we won't be notified about new value. If you'd like to accept only non-optional values you can declare `CockpitParam<T>` as `CockpitParam<T : Any>`